dot/state/block.go

```diff
@@ -937,3 +946,18 @@ func (bs *BlockState) StoreRuntime(hash common.Hash, rt runtime.Instance) {
 func (bs *BlockState) GetNonFinalisedBlocks() []common.Hash {
 	return bs.bt.GetAllBlocks()
 }
+
+func (bs *BlockState) Checkpoint(header *types.Header) {
+	if header.Number < bs.nextCheckpoint {
+		return
+	}
+
+	logger.Infof("creating a checkpoint, latest finalized block %s (#%d)",
+		header.Hash(), header.Number)
+
+	bs.nextCheckpoint = header.Number + checkpointEvery
```

Comment:
Why not?

```suggestion
	bs.nextCheckpoint += checkpointEvery
```

Reply:
This way we don't cover when we start from a snapshot, let's say we start using a snapshot from the block `#2 000 000`, since we started the node `bs.nextCheckpoint` is zero and by `bs.nextCheckpoint += checkpointEvery` will be `1 000 000` so every new block it will take snapshot since the condition to take the snapshot is `header.Number >= bs.nextCheckpoint`

Reply:
Ok I got the point but I think we should solve that case when we are setting the first `nextCheckpoint` instead using the `header.Number` here